src/sql/src/plan/statement/ddl.rs

```diff
@@ -4941,12 +4942,61 @@ pub fn plan_alter_object_rename(
         (ObjectType::ClusterReplica, UnresolvedObjectName::ClusterReplica(name)) => {
             plan_alter_cluster_replica_rename(scx, object_type, name, to_item_name, if_exists)
         }
+        (ObjectType::Schema, UnresolvedObjectName::Schema(name)) => {
+            plan_alter_schema_rename(scx, name, to_item_name, if_exists)
+        }
         (object_type, name) => {
             unreachable!("parser set the wrong object type '{object_type:?}' for name {name:?}")
         }
     }
 }
 
+pub fn plan_alter_schema_rename(
+    scx: &mut StatementContext,
+    name: UnresolvedSchemaName,
+    to_schema_name: Ident,
+    if_exists: bool,
+) -> Result<Plan, PlanError> {
+    let Some((db_spec, schema_spec)) = resolve_schema(scx, name.clone(), if_exists)? else {
+        let object_type = ObjectType::Schema;
+        scx.catalog.add_notice(PlanNotice::ObjectDoesNotExist {
+            name: name.to_ast_string(),
+            object_type,
+        });
+        return Ok(Plan::AlterNoop(AlterNoopPlan { object_type }));
+    };
+
+    // TODO(parkmycar): Make this a structured error and re-evaluate if it needs to be an error.
```

Comment:
This is probably the right error to use: https://github.com/MaterializeInc/materialize/blob/100eacd09d512bf97fca847b90536ef4803785a5/src/adapter/src/catalog/error.rs#L45-L46

Reply:
Refactored this away